### PR TITLE
nixos/hedgedoc: update features note with hedgedoc

### PR DESCRIPTION
**Pull request by SuperSandro2000**
Before the features note wasn't updated when updating hedgedoc

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/web-apps/hedgedoc.nix [hunk 1]
```diff
@@ -299,7 +299,8 @@ in
       };
       docsPath = mkOption {
         type = types.nullOr types.str;
-        default = "./public/docs";
+        default = "${cfg.package}/public/docs";
+        defaultText = "$''{cfg.package}/public/docs";
```
Comment (drupol) [inline nixos/modules/services/web-apps/hedgedoc.nix:303]: Nice ! I've learned something :)

There are also those keys missing:
- `defaultNotePath`
- `indexPath`
- `hackmdPath`

**Review by drupol — APPROVED** [review]
:100:

File: nixos/modules/services/web-apps/hedgedoc.nix [hunk 2]
```diff
@@ -291,58 +291,62 @@ in
       };
       defaultNotePath = mkOption {
         type = types.nullOr types.str;
-        default = "./public/default.md";
+        default = "${cfg.package}/public/default.md";
+        defaultText = "$''{cfg.package}/public/default.md";
```
Comment (ncfavier) [inline nixos/modules/services/web-apps/hedgedoc.nix:295]:
```suggestion
        defaultText = literalExpression "\${cfg.package}/public/default.md";
```

Same for the other options

**Review by ncfavier — CHANGES_REQUESTED** [review]

File: nixos/modules/services/web-apps/hedgedoc.nix (hunk 2 above)
Comment (ncfavier) [reply nixos/modules/services/web-apps/hedgedoc.nix:295]: Wait no, sorry

```suggestion
        defaultText = literalExpression ''"''${cfg.package}/public/default.md"'';
```

Comment (SuperSandro2000) [reply nixos/modules/services/web-apps/hedgedoc.nix:295]: Lets take the second with the escaped ``\"`` because ``''"''`` is a bit ugly.

**Review by drupol — APPROVED** [review]
LGTM now

File: nixos/modules/services/web-apps/hedgedoc.nix [hunk 3]
```diff
@@ -291,58 +291,62 @@ in
...
         description = lib.mdDoc ''
           Path to the default Note file.
           (Non-canonical paths are relative to HedgeDoc's base directory)
         '';
       };
       docsPath = mkOption {
         type = types.nullOr types.str;
-        default = "./public/docs";
+        default = "${cfg.package}/public/docs";
+        defaultText = literalExpression "\"\$''{cfg.package}/public/docs\"";
```
Comment (ncfavier) [inline nixos/modules/services/web-apps/hedgedoc.nix:304]:
```suggestion
        defaultText = literalExpression "\"\${cfg.package}/public/docs\"";
```

File: nixos/modules/services/web-apps/hedgedoc.nix [hunk 4]
```diff
@@ -351,7 +355,7 @@ in
       uploadsPath = mkOption {
         type = types.str;
         default = "${cfg.workDir}/uploads";
-        defaultText = literalExpression "/var/lib/${name}/uploads";
+        defaultText = literalExpression "\"/var/lib/${name}/uploads\"";
```
Comment (ncfavier) [inline nixos/modules/services/web-apps/hedgedoc.nix:358]:
```suggestion
        defaultText = literalExpression "\"\${cfg.workDir}/uploads\"";
```

**Review by drupol — APPROVED** [review]

File: nixos/modules/services/web-apps/hedgedoc.nix (hunk 3 above)
Comment (ncfavier) [reply nixos/modules/services/web-apps/hedgedoc.nix]: Please fix all the instances of this

File: nixos/modules/services/web-apps/hedgedoc.nix [hunk 5]
```diff
@@ -291,58 +291,62 @@ in
...
         description = lib.mdDoc ''
           Path to the docs directory.
           (Non-canonical paths are relative to HedgeDoc's base directory)
         '';
       };
       indexPath = mkOption {
         type = types.nullOr types.str;
-        default = "./public/views/index.ejs";
+        default = "${cfg.package}/public/views/index.ejs";
+        defaultText = literalExpression "\"\$''{cfg.package}/public/views/index.ejs\"";
```
Comment (ncfavier) [inline nixos/modules/services/web-apps/hedgedoc.nix:313]:
```suggestion
        defaultText = literalExpression "\"\${cfg.package}/public/views/index.ejs\"";
```

File: nixos/modules/services/web-apps/hedgedoc.nix [hunk 6]
```diff
@@ -291,58 +291,62 @@ in
...
         description = lib.mdDoc ''
           Path to the index template file.
           (Non-canonical paths are relative to HedgeDoc's base directory)
         '';
       };
       hackmdPath = mkOption {
         type = types.nullOr types.str;
-        default = "./public/views/hackmd.ejs";
+        default = "${cfg.package}/public/views/hackmd.ejs";
+        defaultText = "$''{cfg.package}/public/views/hackmd.ejs";
```
Comment (ncfavier) [inline nixos/modules/services/web-apps/hedgedoc.nix:322]:
```suggestion
        defaultText = literalExpression "\"\${cfg.package}/public/views/hackmd.ejs\"";
```

File: nixos/modules/services/web-apps/hedgedoc.nix [hunk 7]
```diff
@@ -291,58 +291,62 @@ in
...
           Path to the hackmd template file.
           (Non-canonical paths are relative to HedgeDoc's base directory)
         '';
       };
       errorPath = mkOption {
         type = types.nullOr types.str;
-        default = null;
-        defaultText = literalExpression "./public/views/error.ejs";
+        default = "${cfg.package}/public/views/error.ejs";
+        defaultText = literalExpression "\"\$''{cfg.package}/public/views/error.ejs\"";
```
Comment (ncfavier) [inline nixos/modules/services/web-apps/hedgedoc.nix:331]:
```suggestion
        defaultText = literalExpression "\"\${cfg.package}/public/views/error.ejs\"";
```

File: nixos/modules/services/web-apps/hedgedoc.nix [hunk 8]
```diff
@@ -291,58 +291,62 @@ in
...
           Path to the error template file.
           (Non-canonical paths are relative to HedgeDoc's base directory)
         '';
       };
       prettyPath = mkOption {
         type = types.nullOr types.str;
-        default = null;
-        defaultText = literalExpression "./public/views/pretty.ejs";
+        default = "${cfg.package}/public/views/pretty.ejs";
+        defaultText = literalExpression "\"\$''{cfg.package}/public/views/pretty.ejs\"";
```
Comment (ncfavier) [inline nixos/modules/services/web-apps/hedgedoc.nix:340]:
```suggestion
        defaultText = literalExpression "\"\${cfg.package}/public/views/pretty.ejs\"";
```

File: nixos/modules/services/web-apps/hedgedoc.nix [hunk 9]
```diff
@@ -291,58 +291,62 @@ in
...
           Path to the pretty template file.
           (Non-canonical paths are relative to HedgeDoc's base directory)
         '';
       };
       slidePath = mkOption {
         type = types.nullOr types.str;
-        default = null;
-        defaultText = literalExpression "./public/views/slide.hbs";
+        default = "${cfg.package}/public/views/slide.hbs";
+        defaultText = literalExpression "\"\$''{cfg.package}/public/views/slide.hbs\"";
```
Comment (ncfavier) [inline nixos/modules/services/web-apps/hedgedoc.nix:349]:
```suggestion
        defaultText = literalExpression "\"\${cfg.package}/public/views/slide.hbs\"";
```

**Review by ncfavier — APPROVED** [review]

**Pull request merged**